nav2_costmap_2d/src/layered_costmap.cpp

```diff
@@ -80,6 +90,12 @@ LayeredCostmap::LayeredCostmap(std::string global_frame, bool rolling_window, bo
   }
 }
 
+LayeredCostmap::LayeredCostmap(std::string global_frame, bool rolling_window, bool track_unknown)
+: LayeredCostmap(rclcpp::get_logger("nav2_costmap_2d"), rclcpp::Clock{RCL_ROS_TIME}, global_frame,
```

Comment:
I don't think there's a sane default for the clock type like this. I think this needs to always be a real clock if a clock is used. That's an argument - to me - to change the constructor of LayeredCostmap to have the clock/logger fields at the end in the existing constructor rather than adding a new one. If we're adding it, make it strictly required since there's no sane workaround.